### ISSUE TITLE
"Expand macro at caret" uses `$crate` instead of `std`

### ISSUE THREAD

**Issue by lnicola** [rust-lang/rust-analyzer#16438]
```rust
#[derive(Debug)]
struct S;

// ->

impl $crate::fmt::Debug for S {
    fn fmt(&self, f: &mut $crate::fmt::Formatter) -> $crate::fmt::Result {
        match self {
            S => f.write_str("S"),
        }
    }
}
```

This doesn't seem to happen every time though, in rust-lang/rustc it used `::std`.

**Comment by Veykril** [comment]
`expand macro at caret` generally still doesn't replace `$crate`, and we now switched to the builtin derives to use that as well (with core/std crate hygiene).

**Comment by lnicola** [comment]
Fixed in #18131.
